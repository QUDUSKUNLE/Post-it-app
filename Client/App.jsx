import React from 'react';
import Home from './components/Home';
import NavLink from './components/NavLink';
import { Link, IndexLink } from 'react-router-dom';

class App extends React.Component{
	render() {
		return (
			<div>
				<nav className="black-text" role="navigation">
					<div className="nav-wrapper">
						<Link to="/" className="brand-logo">PostIt<small>App</small></Link>
						<ul className="right hide-on-med-and-down">
							<li><IndexLink to="/" activeClassName="active">Home</IndexLink></li>
							<li><NavLink to="/signup">userSignUp</NavLink></li>
							<li><NavLink to="/signin">userSignIn</NavLink></li>
						</ul>
					</div>
				</nav>
				{this.props.children || <Home />}
			</div>
		);
	}
};

export default App;
