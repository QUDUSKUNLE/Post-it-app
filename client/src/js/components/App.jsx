import React from 'react';
import Navbar from './navBar.jsx';
import Home from './Home.jsx';

/**
  * Represents App Component.
*/
class App extends React.Component {
  /**
    * @override
  */
  render() {
    return (
      <div>
        <Navbar />
        <Home />
      </div>
    );
  }
}

export default App;
