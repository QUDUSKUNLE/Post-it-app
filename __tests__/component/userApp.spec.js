import React from 'react';
import { shallow } from 'enzyme';
import App from '../../client/src/js/components/userApp.jsx';
import SignUp from '../../client/src/js/components/userSignUp.jsx';

describe('PostIt-app', () => {
  it('expects App component to be defined', () => {
    expect(App).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const component = shallow(<SignUp/>);
    expect(component).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    const component = shallow(<App/>);
    expect(component.find('.home').exists()).toBeTruthy();
  });
});