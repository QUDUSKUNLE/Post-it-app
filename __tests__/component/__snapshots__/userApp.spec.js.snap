// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostIt-app should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <NavBar />
    <div
      className="container"
    >
      <div
        className="row home"
      >
        <div
          className="col-md-6"
        >
          <h4>
            PostIt
            <small>
              <i>
                App
              </i>
            </small>
              allows friends to come together and share vital informtion.
          </h4>
        </div>
        <SignUp />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <NavBar />
      <div
        className="container"
      >
        <div
          className="row home"
        >
          <div
            className="col-md-6"
          >
            <h4>
              PostIt
              <small>
                <i>
                  App
                </i>
              </small>
                allows friends to come together and share vital informtion.
            </h4>
          </div>
          <SignUp />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <NavBar />
          <div
            className="container"
          >
            <div
              className="row home"
            >
              <div
                className="col-md-6"
              >
                <h4>
                  PostIt
                  <small>
                    <i>
                      App
                    </i>
                  </small>
                    allows friends to come together and share vital informtion.
                </h4>
              </div>
              <SignUp />
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <NavBar />
          <div
            className="container"
          >
            <div
              className="row home"
            >
              <div
                className="col-md-6"
              >
                <h4>
                  PostIt
                  <small>
                    <i>
                      App
                    </i>
                  </small>
                    allows friends to come together and share vital informtion.
                </h4>
              </div>
              <SignUp />
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`PostIt-app should render correctly 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="col-md-5 col-md-offset-1"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 w3-card w3-white"
          id="signupform"
        >
          <h4>
            Create an account
          </h4>
          <br />
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="signinform inp"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="johndoe@example.com"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="username"
              >
                Username
              </label>
              <input
                className="signinform"
                id="username"
                name="username"
                onChange={[Function]}
                placeholder="johndoe"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="password"
              >
                Password
              </label>
              <input
                className="signinform"
                id="pass"
                name="password"
                onChange={[Function]}
                placeholder="********"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="conf_password"
              >
                Confirm Password
              </label>
              <input
                className="signinform"
                id="conf_password"
                name="conf_password"
                onChange={[Function]}
                placeholder="********"
                required={true}
                type="password"
                value=""
              />
            </div>
            <button
              className="signinformbtn"
              type="submit"
            >
              Sign up
            </button>
          </form>
          <br />
          <p
            className="text-center"
          >
            By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="col-md-5 col-md-offset-1"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 w3-card w3-white"
            id="signupform"
          >
            <h4>
              Create an account
            </h4>
            <br />
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Email
                </label>
                <input
                  className="signinform inp"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="johndoe@example.com"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="username"
                >
                  Username
                </label>
                <input
                  className="signinform"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="johndoe"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="password"
                >
                  Password
                </label>
                <input
                  className="signinform"
                  id="pass"
                  name="password"
                  onChange={[Function]}
                  placeholder="********"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="conf_password"
                >
                  Confirm Password
                </label>
                <input
                  className="signinform"
                  id="conf_password"
                  name="conf_password"
                  onChange={[Function]}
                  placeholder="********"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <button
                className="signinformbtn"
                type="submit"
              >
                Sign up
              </button>
            </form>
            <br />
            <p
              className="text-center"
            >
              By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
            </p>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignUp />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignUp {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "conf_password": "",
          "email": "",
          "errMessage": "",
          "password": "",
          "signupMessage": "",
          "username": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            className="col-md-5 col-md-offset-1"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 w3-card w3-white"
                id="signupform"
              >
                <h4>
                  Create an account
                </h4>
                <br />
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="signinform inp"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                    <input
                      className="signinform"
                      id="username"
                      name="username"
                      onChange={[Function]}
                      placeholder="johndoe"
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      className="signinform"
                      id="pass"
                      name="password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="conf_password"
                    >
                      Confirm Password
                    </label>
                    <input
                      className="signinform"
                      id="conf_password"
                      name="conf_password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Sign up
                  </button>
                </form>
                <br />
                <p
                  className="text-center"
                >
                  By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
                </p>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <div
            className="col-md-5 col-md-offset-1"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 w3-card w3-white"
                id="signupform"
              >
                <h4>
                  Create an account
                </h4>
                <br />
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="signinform inp"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                    <input
                      className="signinform"
                      id="username"
                      name="username"
                      onChange={[Function]}
                      placeholder="johndoe"
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      className="signinform"
                      id="pass"
                      name="password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="conf_password"
                    >
                      Confirm Password
                    </label>
                    <input
                      className="signinform"
                      id="conf_password"
                      name="conf_password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Sign up
                  </button>
                </form>
                <br />
                <p
                  className="text-center"
                >
                  By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
                </p>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignUp />,
}
`;
